cal/cal/static/js/services.js

```diff
@@ -214,3 +214,30 @@ analyticsApp.service("CategoryService", ['$http', '$q', function($http, $q) {
   };
 
 }]);
+
+analyticsApp.service('QueryService', function() {
+  var ctrlDetails = [];
```

Comment:
I don't think you need this? Declare the variable inside the `populateData` function